action.yaml

```diff
@@ -0,0 +1,121 @@
+    required: false
+    default: ""
+  allowlist_file_path:
+    description: "Path to allowlist.json file"
+    required: false
+    default: ""
+  from_tag:
+    description: "Starting tag to scan from"
+    required: false
+    default: ""
+  to_tag:
+    description: "Ending tag to scan to"
+    required: false
+    default: ""
+# Internal Variables. Not meant to be passed by the user. These are set up by the action itself.
+  github_api_url:
+    description: "Base URL of Github API"
+    required: false
+    default: ${{ github.api_url }}
+  github_repository:
+    description: "Repository name"
+    required: false
+    default: ${{ github.repository }}
+  github_token:
+    description: "Github Token"
+    required: false
+    default: ${{ github.token }}
+  github_pull_request:
+    description: "Github PR"
+    required: false
+    default: ${{ github.event.number }}
+  github_repo_url:
+    description: "Github Repo URL"
+    required: false
+    default: ${{ github.repositoryUrl }}
+  github_repo_owner:
+    description: "Github Repo Owner"
+    required: false
+    default: ${{ github.repository_owner }}
+
+runs:
+  using: "composite"
+  steps:
+
+    - name: Docker Setup QEMU
+      uses: docker/setup-qemu-action@v3.1.0
+
+    - name: Run Fossology scan in Docker
+      shell: bash
+      run: |
+        # Prepare docker run command with arguments
+        docker_cmd="docker run --rm --name fossologyscanner -w /opt/repo -v ${PWD}:/opt/repo \
+            -e GITHUB_TOKEN=${{ inputs.github_token }} \
+            -e GITHUB_PULL_REQUEST=${{ inputs.github_pull_request }} \
+            -e GITHUB_REPOSITORY=${{ inputs.github_repository }} \
+            -e GITHUB_API=${{ inputs.github_api_url }} \
+            -e GITHUB_REPO_URL=${{ inputs.github_repo_url }} \
+            -e GITHUB_REPO_OWNER=${{ inputs.github_repo_owner }} \
+            -e GITHUB_ACTIONS"
+        
+        if [ "$KEYWORD_CONF_FILE_PATH" != "" ]; then
+        docker_cmd+=" -v ${{ github.workspace }}/${{ inputs.keyword_conf_file_path }}:/bin/${{ inputs.keyword_conf_file_path}}"
+        fi
+        if [ "${{ inputs.allowlist_file_path }}" != "" ]; then
+        docker_cmd+=" -v ${{ github.workspace }}/${{ inputs.allowlist_file_path}}:/bin/${{ inputs.allowlist_file_path }}"
+        fi
+
+        docker_cmd+=" fossology/fossology:scanner /bin/fossologyscanner"
+        docker_cmd+=" ${{ inputs.scanners }}"
+        docker_cmd+=" ${{ inputs.scan_mode }}"
+
+        # Add additional conditions
+        if [ "${{ inputs.scan_mode }}" == "differential" ]; then
+        docker_cmd+=" --tags ${{ inputs.from_tag }} ${{ inputs.to_tag }}"
+        fi
+        if [ "${{ inputs.keyword_conf_file_path }}" != "" ]; then
+        docker_cmd+=" --keyword-conf ${{ inputs.keyword_conf_file_path }}"
+        fi
+        if [ "${{ inputs.allowlist_file_path }}" != "" ]; then
+        docker_cmd+=" --allowlist-path ${{ inputs.allowlist_file_path }}
```

Comment:
```suggestion
        docker_cmd+=" --allowlist-path ${{ inputs.allowlist_file_path }}"
```
I think you should move this to the shell script so that linting errors can be fixed beforehand. IMO, it would be easier to maintain a separate script file.

Reply:
At first, I wrote it in the script.sh file itself, but during execution the runner was not able to find that file for some reason.

<img width="958" alt="Screenshot 2024-07-17 at 11 17 24 PM" src="https://github.com/user-attachments/assets/d7d944d8-f64a-4800-b49c-72050660390a">

I wrote the changes in script.sh and `action.yaml` looked like this
<img width="448" alt="Screenshot 2024-07-17 at 11 19 16 PM" src="https://github.com/user-attachments/assets/7b4a2d8f-dfdd-490d-bddf-f2d2e337e9c7">

Reply:
@avinal are you okay with changes suggested here?

Reply:
@avinal 
I wrote the script logic in a separate file and I have committed by changing the file permissions as `755`.
Checking out the actions repository wasn't necessary since we have an environment variable `GITHUB_ACTION_PATH` to access the directory where the GH action runner stores all the actions repositories being used. 
PS. Before tagging and releasing, we might need to update the permissions of the file again as I am not sure whether it would persist after the merge commit or not.